### DISCUSSION TITLE
[@types/inquirer-autocomplete-prompt] Module declaration

### DISCUSSION THREAD

**Discussion by ipatalas** [DefinitelyTyped/DefinitelyTyped#63815] [Issues with a @types package]
Hi,

I'm wondering what was the decision behind not putting the following code into main `index.d.ts` file:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/75c4ee2f1e60edd885464025baa01c4026eb7299/types/inquirer-autocomplete-prompt/inquirer-autocomplete-prompt-tests.ts#L6-L10

I do realize this autocomplete plugin is registered at runtime, however even if that declaration is included you have to at least import autocomplete plugin to make it available:
![image](https://user-images.githubusercontent.com/2623329/210880075-74beb84c-d955-4c7b-8071-c0c866294717.png)

![image](https://user-images.githubusercontent.com/2623329/210880122-c6072757-1e6d-4f6d-b50c-b5dcf6ca32d1.png)

That's not very handy not to have it included as I use inquirer across many different files and I'd need to include that declaration in each of them.

Including it is not perfect, because it will still suggest the typings even if you don't physically register `inquirer.registerPrompt('autocomplete' ...)` but I think it's not such a big harm, what do you think?

I'm not sure this is even possible in Typescript to enrich a type at runtime. Anyone knows how does other libraries with runtime registered plugins approach this?

[a comment by an automated account is not shown]

**Comment by jayeeson** [comment]
Makes sense to add it. I think the downside you mention is not such a big harm indeed.

**Reply by ipatalas** [reply]
All right, I can do that + update some of the typings as they are outdated. 
However I'm having hard time running the tests. I did [partial checkout](https://github.com/DefinitelyTyped/DefinitelyTyped/#partial-clone) getting both `inquirer-autocomplete-prompt` and `inquirer` itself.
Then I run `npm test inquirer-autocomplete-prompt` and getting errors:
```shell
> definitely-typed@0.0.3 test
> node --require source-map-support/register node_modules/@definitelytyped/dtslint/ types "inquirer-autocomplete-prompt"

dtslint@0.0.142
Error: 
A module look-up failed, this often occurs when you need to run `npm install` on a dependent module before you can lint.

Before you debug, first try running:

   npm install --prefix types/inquirer

../inquirer/lib/prompts/expand.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/prompts/input.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/prompts/list.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/prompts/number.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/prompts/password.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/prompts/rawlist.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/ui/baseUI.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/utils/events.d.ts(1,53): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/utils/readline.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.
../inquirer/lib/utils/screen-manager.d.ts(1,48): error TS2307: Cannot find module 'readline' or its corresponding type declarations.

    at testTypesVersion (...\node_modules\@definitelytyped\dtslint\src\index.ts:235:11)
    at runTests (...\node_modules\@definitelytyped\dtslint\src\index.ts:197:7)
    at main (...\node_modules\@definitelytyped\dtslint\src\index.ts:83:5)
```

All errors seems to be related to `inquirer` itself but when I run tests for `inquirer` only I get this:
```shell
> definitely-typed@0.0.3 test
> node --require source-map-support/register node_modules/@definitelytyped/dtslint/ types "inquirer"

dtslint@0.0.142
Error: Errors in typescript@5.0 for external dependencies:
node_modules/rxjs/dist/types/internal/scheduler/timerHandle.d.ts(1,62): error TS2304: Cannot find name 'setTimeout'.
```

According to README it should be rather simple but it doesn't work. Tried one more other library and similar story 🤷🏻‍♂️ This is my first time trying to do something here, any clue?

**Reply by jayeeson** [reply]
You are missing modules, maybe you have not yet run `npm install`.

**Reply by ipatalas** [reply]
Well, I actually did run `npm install`. I tried starting over and here is full command history:
```shell
git clone --sparse --filter=blob:none --depth=1 https://github.com/DefinitelyTyped/DefinitelyTyped.git
cd DefinitelyTyped
git sparse-checkout add types/inquirer-autocomplete-prompt types/inquirer
npm install
npm test inquirer-autocomplete-prompt # or npm test inquirer
```

Still both of them fail.

Running `npm test inquirer-autocomplete-prompt` gives a suggestion:
```
 D:\Dev\src\DefinitelyTyped  master ±2  definitely-typed@0.0.3                                            11:27:43 
λ npm test inquirer-autocomplete-prompt

> definitely-typed@0.0.3 test
> node --require source-map-support/register node_modules/@definitelytyped/dtslint/ types "inquirer-autocomplete-prompt"

dtslint@0.0.142
Error:
A module look-up failed, this often occurs when you need to run `npm install` on a dependent module before you can lint.

Before you debug, first try running:

   npm install --prefix D:/Dev/src/DefinitelyTyped/types/inquirer

Then re-run. Full error logs are below.
```

I did that and it doesn't help :(

**Reply by jayeeson** [reply]
I think something is messed up with the partial checkout. I experience the same issues.

It's working with full checkout though.... it's only 1.6GB ;)

**Reply by ipatalas** [reply]
I was hoping to avoid that but well, if it works then I'm convinced, thx! :)
